packages/react/src/components/header/components/headerLanguageSelector/HeaderLanguageSelector.tsx

```diff
@@ -48,61 +38,70 @@ const renderLanguageNode = (language: LanguageOption) => {
   return <LanguageButton key={language.value} value={language.value} label={language.label} />;
 };
 
-export const HeaderLanguageSelector = ({
+export function HeaderLanguageSelector(props: LanguageSelectorProps) {
+  // This is a dummy component used for passing language selector props and children.
+  // Those are extracted in <HeaderActionBar /> and rendered with <HeaderLanguageSelectorConsumer />
+
+  // use props in some way to avoid ts-error
+  return props ? null : null;
+}
```

Comment:
This feels a bit hacky but I can't think of a better solution with my brains already in vacation mode.